From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Chris Thompson <cthomp@chromium.org>
Date: Tue, 6 Jan 2026 10:59:59 -0800
Subject: [PATCH] disable unsupported HTTPS-first dialog UI by default in
 Android

upstream original commit name:
[HFM] Disable dialog UI flag on Android

The `HttpsFirstDialogUi` flag was default enabled in
crrev.com/c/7173972, but this had unintended side-effects on Android.
For expediency, this CL just disables the feature flag on Android.

Bug: 469092867
Change-Id: I9d39b7ec1aa918c681a290ea1fdbf24d73dfac32
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/7399299
Reviewed-by: Mustafa Emre Acer <meacer@chromium.org>
Auto-Submit: Chris Thompson <cthomp@chromium.org>
Commit-Queue: Chris Thompson <cthomp@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1565119}
---
 components/security_interstitials/core/features.cc | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/components/security_interstitials/core/features.cc b/components/security_interstitials/core/features.cc
index 5a5d29fa26508..5fa91c3acf070 100644
--- a/components/security_interstitials/core/features.cc
+++ b/components/security_interstitials/core/features.cc
@@ -5,10 +5,18 @@
 #include "components/security_interstitials/core/features.h"
 
 #include "base/feature_list.h"
+#include "build/build_config.h"
 
 namespace security_interstitials::features {
 
 // Enables a dialog-based UI for HTTPS-First Mode.
+// The flag is currently disabled on Android as the new UI is not implemented
+// there yet. (See crbug.com/469092867 for some previous unintended side
+// effects.)
+#if BUILDFLAG(IS_ANDROID)
+BASE_FEATURE(kHttpsFirstDialogUi, base::FEATURE_DISABLED_BY_DEFAULT);
+#else
 BASE_FEATURE(kHttpsFirstDialogUi, base::FEATURE_ENABLED_BY_DEFAULT);
+#endif
 
 }  // namespace security_interstitials::features
